os/arch/arm/src/armv7-a/arm_assert.c

```diff
@@ -0,0 +1,985 @@
+
+	rewind(pFile);
+	mid = (first + last) / 2;
+	while (first <= last) {
+		fseek(pFile, mid, SEEK_SET);
+
+		/* If the file pointer is in the mid of the line, make sure
+		 * it's been properly moved to start of next line
+		 */
+		for (c = getc(pFile); c != '\n'; c = getc(pFile)) {
+			if (c == EOF) {
+				lldbg("Reached end of file and couldn't find symbol\n");
+				fclose(pFile);
+				return -1;
+			}
+		}
+
+		word = 0;
+		// Read 2 lines and Split the string as words
+		for (read_line = 0; read_line < 2; read_line++) {
+			fgets(line, 128, pFile);
+			ch_in_line = 0;
+			ch_in_word = 0;
+			while (line[ch_in_line] != '\0') {
+				if (line[ch_in_line] != ' ') {
+					data[word][ch_in_word++] = line[ch_in_line];
+				} else {
+					data[word][ch_in_word] = '\0';
+					word++;
+					ch_in_word = 0;
+				}
+				ch_in_line++;
+				if (line[ch_in_line] == '\0') {
+					data[word][ch_in_word] = '\0';
+					word++;
+					ch_in_word = 0;
+				}
+			}
+		}
+		/* Convert the string data to hexadecimal */
+		addr = strtoul(data[0], NULL, 16);
+		next_addr = strtoul(data[3], NULL, 16);
+		if (search_addr >= addr && search_addr < next_addr) {
+			total = snprintf(buffer, buflen, "%s", data[2]);
+			sym_offset = search_addr - addr;
+			total_size = next_addr - addr;
+			snprintf(&buffer[total - 1], buflen - total, "+0x%lx/0x%lx", sym_offset, total_size);
+			break;
+		}
+		if (search_addr < addr) {
+			last = mid - 1;
+		} else {
+			first = mid + 1;
+		}
+
+		mid = (first + last) / 2;
+	}
+	if (first > last) {
+		lldbg("symbol is not found in system map\n");
+		buffer = "";
+	}
+
+	/* Close the file */
+	fclose(pFile);
+
+	return 0;
+}
+#endif
+
+/****************************************************************************
+ * Name: unwind_frame_with_fp
+ ****************************************************************************/
+
+static int unwind_frame_with_fp(struct stackframe *stack_frame, uint32_t stacksize)
+{
+	//Dummy
+	return 0;
+	uint32_t high, low, stack_align_mask;
+	uint32_t fp = stack_frame->framePointer;	/* Read the frame pointer */
+
+	low = stack_frame->stackPointer;	/* Read the Stack pointer from frame */
+	stack_align_mask = stacksize - 1;
+	high = (low + stack_align_mask) & ~(stack_align_mask);
+
+	/* lldbg("\n %s : low : 0x%p mask: 0x%p and high: 0x%p\n", __func__, low, mask, high); */
+
+	/*
+	 * check whether fp is  within the lower and higher stack boundary
+	 */
+	if (fp < low + 12 || fp > high - 4) {
+		lldbg("End of Callstack\n");
+		return -1;
+	}
+
+	/* Restore registers from stack frame */
+	stack_frame->programCounter = MEM_READ(fp - 4);
+	stack_frame->stackPointer = MEM_READ(fp - 8);
+	stack_frame->framePointer = MEM_READ(fp - 12);
+
+	return 0;
+}
+
+/****************************************************************************
+ * Name: unwind_backtrace_with_fp
+ ****************************************************************************/
+
+static void unwind_backtrace_with_fp(arm_regs_t *regs, struct tcb_s *task)
```

Comment:
Same as above.
Please check if arch supports stack frame variables and if this API is needed.